packages/cadl-ranch-specs/http/azure/core/lro/rpc-legacy/mockapi.ts

```diff
@@ -25,3 +25,44 @@ Scenarios.Azure_Core_Lro_Rpc_Legacy_CreateResourcePollViaOperationLocation = pas
     return { status: 200, body: json(response) };
   }),
 ]);
+
+const expectedModel = { modelId: "123", description: "hello" };
+let createPollResourceLocationCount = 0;
+
+Scenarios.Azure_Core_Lro_Rpc_Legacy_CreateResourcePollViaOperationLocationAndResourceLocation = passOnSuccess([
+  mockapi.post(
+    "/azure/core/lro/rpc/legacy/create-resource-poll-via-operation-location-final-result-in-resource-location/documentModels:build",
+    (req) => {
+      req.expect.containsQueryParam("api-version", "2022-12-01-preview");
+      req.expect.bodyEquals(expectedModel);
+      createPollResourceLocationCount = 0;
+      return {
+        status: 202,
+        headers: {
+          "operation-location": `${req.baseUrl}/azure/core/lro/rpc/legacy/create-resource-poll-via-operation-location-final-result-in-resource-location/operations/1234567890?api-version=2022-12-01-preview`,
+        },
+      };
+    },
+  ),
+  mockapi.get(
+    "/azure/core/lro/rpc/legacy/create-resource-poll-via-operation-location-final-result-in-resource-location/operations/1234567890",
+    (req) => {
+      req.expect.containsQueryParam("api-version", "2022-12-01-preview");
+      const resourceLocation = `${req.baseUrl}/azure/core/lro/rpc/legacy/create-resource-poll-via-operation-location-final-result-in-resource-location/documentModels/123?api-version=2022-12-01-preview`;
+      const baseResponse = { operationId: "1234567890", resourceLocation: resourceLocation };
+      const response =
+        createPollResourceLocationCount > 0
+          ? { ...baseResponse, status: "succeeded" }
+          : { ...baseResponse, status: "notStarted" };
+      createPollResourceLocationCount += 1;
+      return { status: 200, body: json(response) };
+    },
+  ),
+  mockapi.get(
+    "/azure/core/lro/rpc/legacy/create-resource-poll-via-operation-location-final-result-in-resource-location/documentModels/123",
+    (req) => {
+      req.expect.containsQueryParam("api-version", "2022-12-01-preview");
+      return { status: 200, body: json(expectedModel) };
+    },
+  ),
```

Comment:
Not sure whether this is required to pass the test.

We should not require this, as if SDK follow the standard "result" pattern, it would also be OK. (though a hack in test could just call this individually...)

Reply:
Yeah, maybe it is not necessary for some language, but I don't find way to  mark it as optional. For now, language can call it individually for now as you said

Reply:
Any suggestion ? @lirenhe